web/src/main/java/org/mskcc/cbio/oncokb/api/pub/v1/AnnotationsApiController.java

```
@@ -706,4 +544,236 @@ private IndicatorQueryResp getIndicatorQueryFromHGVSg(
+        if (sample.getMutations() != null) {
+            if (sample.getMutations().getGenomicChange() != null) {
+                genomicChange = annotateMutationsByGenomicChange(sample.getMutations().getGenomicChange());
+            }
+            if (sample.getMutations().getProteinChange() != null) {
+                proteinChange = annotateMutationsByProteinChange(sample.getMutations().getProteinChange());
+            }
+            if (sample.getMutations().getHgvsg() != null) {
+                hgvsg = annotateMutationsByHGVS(sample.getMutations().getHgvsg());
+            }
+            if (sample.getMutations().getcDnaChange() != null) {
+                cDnaChange = annotateMutationsByHGVS(sample.getMutations().getcDnaChange());
+            }
+        }
+
+        annotatedSample.setStructuralVariants(structuralVariants);
+        annotatedSample.setCopyNumberAlterations(copyNumberAlterations);
+        annotatedSample.setMutations(Stream.of(genomicChange, cDnaChange, proteinChange, hgvsg)
+            .flatMap(Collection::stream)
+            .collect(Collectors.toList())
+        );
+        return annotatedSample;
+    }
+
+    private List<IndicatorQueryResp> annotateStructuralVariants(List<AnnotateStructuralVariantQuery> structuralVariants) {
+        List<IndicatorQueryResp> result = new ArrayList<>();
+        for (AnnotateStructuralVariantQuery query : structuralVariants) {
+            Gene geneA = new Gene();
+            if (query.getGeneA() != null) {
+                try {
+                    geneA = this.cacheFetcher.findGeneBySymbol(
+                        query.getGeneA().getEntrezGeneId() != null ?
+                            query.getGeneA().getEntrezGeneId().toString() :
+                            query.getGeneA().getHugoSymbol()
+                    );
+                    if (geneA == null) {
+                        geneA = new Gene();
+                    }
+                } catch (ApiException e) {
+                }
```

Shouldn't this be a 5xx error?

---

I copied this logic over from our existing logic, but it seems to make sense. Sine we are iterating through a list of structural variants, we do not want to return an error, since many of the annotations may succeed. The downstream logic then handles this by returning empty fields for the annotation.